src/middleware/timing/timing.ts

```diff
@@ -30,7 +30,9 @@ interface TimingOptions {
 const getTime = (): number => {
   try {
     return performance.now()
-  } catch {}
+  } catch (e) {
+    console.warn('performance.now() is not available. Falling back to Date.now()', e)
```

Comment:
There are already several `console.warn()` in this file, but these are displayed when “middleware is not set up correctly”.

This fallback to `Date.now()` is displayed when running in an environment where `performance.now()` is not available, but I don't think users would be happy if they were shown a warning for every request.

If we display a warning when it is not available, it would be better to do it like this.

```ts
diff --git a/src/middleware/timing/timing.ts b/src/middleware/timing/timing.ts
index d387f43b..f341592f 100644
--- a/src/middleware/timing/timing.ts
+++ b/src/middleware/timing/timing.ts
@@ -27,12 +27,9 @@ interface TimingOptions {
   crossOrigin?: boolean | string | ((c: Context) => boolean | string)
 }
 
-const getTime = (): number => {
-  try {
-    return performance.now()
-  } catch {}
-  return Date.now()
-}
+const getTime: () => number =
+  performance?.now?.bind(performance) ??
+  (console.warn('performance.now is not supported, using Date.now'), Date.now.bind(Date))
 
 /**
  * Server-Timing Middleware for Hono.
```

Reply:
Hello! thanks for you response

Now I really think that was unnecessary, was planning to add some outputs about possible bugs, but thanks for your comments!


> There are already several `console.warn()` in this file, but these are displayed when “middleware is not set up correctly”.
> 
> This fallback to `Date.now()` is displayed when running in an environment where `performance.now()` is not available, but I don't think users would be happy if they were shown a warning for every request.
> 
> If we display a warning when it is not available, it would be better to do it like this.
> 
> ```ts
> diff --git a/src/middleware/timing/timing.ts b/src/middleware/timing/timing.ts
> index d387f43b..f341592f 100644
> --- a/src/middleware/timing/timing.ts
> +++ b/src/middleware/timing/timing.ts
> @@ -27,12 +27,9 @@ interface TimingOptions {
>    crossOrigin?: boolean | string | ((c: Context) => boolean | string)
>  }
>  
> -const getTime = (): number => {
> -  try {
> -    return performance.now()
> -  } catch {}
> -  return Date.now()
> -}
> +const getTime: () => number =
> +  performance?.now?.bind(performance) ??
> +  (console.warn('performance.now is not supported, using Date.now'), Date.now.bind(Date))
>  
>  /**
>   * Server-Timing Middleware for Hono.
> ```